scene/gui/control.cpp

```diff
@@ -3259,6 +3264,7 @@ void Control::_notification(int p_notification) {
 				data.RI = viewport->_gui_add_root_control(this);
 
 				get_parent()->connect(SNAME("child_order_changed"), callable_mp(get_viewport(), &Viewport::gui_set_root_order_dirty), CONNECT_REFERENCE_COUNTED);
+				ProjectSettings::get_singleton()->connect("settings_changed", callable_mp(this, &Control::_project_settings_changed));
```

Comment:
Surround the signal connection with `if (Engine::get_singleton()->is_editor_hint())`, as this isn't needed outside the editor.